beater/beater.go

```diff
@@ -159,14 +158,17 @@ func (bt *beater) listen() (net.Listener, error) {
 }
 
 func (bt *beater) Run(b *beat.Beat) error {
-	tracer, traceListener, err := initTracer(b.Info, bt.config, bt.logger)
+	tracer, tracerServer, err := initTracer(b.Info, bt.config, bt.logger)
 	if err != nil {
 		return err
 	}
-	if traceListener != nil {
-		defer traceListener.Close()
+	if tracerServer != nil {
+		go func() {
+			defer tracerServer.stop()
+			defer tracer.Close()
```

Comment:
Do I understand this correct that the `tracer` is now only closed for in-process listener (== `tracerServer!= nil`)? Could you explain why?

Reply:
Oops, that was a mistake - thanks, fixed.